pkg/validations/createcluster/createcluster.go

```diff
@@ -0,0 +1,86 @@
+package createcluster
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/gitops/flux"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/validations"
+	"github.com/aws/eks-anywhere/pkg/workflows/interfaces"
+)
+
+type validationManager struct {
+	clusterSpec    *cluster.Spec
+	provider       providers.Provider
+	gitOpsFlux     *flux.Flux
+	cmdValidations interfaces.Validator
+	dockerExec     validations.DockerExecutable
+}
+
+func NewValidationManager(clusterSpec *cluster.Spec, provider providers.Provider, gitOpsFlux *flux.Flux, cmdValidations interfaces.Validator, dockerExec validations.DockerExecutable) *validationManager {
+	return &validationManager{
+		clusterSpec:    clusterSpec,
+		provider:       provider,
+		gitOpsFlux:     gitOpsFlux,
+		cmdValidations: cmdValidations,
+		dockerExec:     dockerExec,
+	}
+}
+
+func (v *validationManager) ValidateCreate(ctx context.Context) error {
+	runner := validations.NewRunner()
+	runner.Register(v.generateCreateValidations(ctx)...)
+	runner.Register(v.gitOpsFlux.Validations(ctx, v.clusterSpec)...)
+	err := runner.Run()
+
+	return err
+}
+
+func (v *validationManager) generateCreateValidations(ctx context.Context) []validations.Validation {
+	return []validations.Validation{
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate docker executable",
+				Err:    validations.ValidateDockerExecutable(ctx, v.dockerExec, runtime.GOOS),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate kubeconfig path",
+				Err:    kubeconfig.ValidateKubeconfigPath(v.clusterSpec.Cluster.Name),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate cluster",
+				Err:    cluster.ValidateConfig(v.clusterSpec.Config),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate supported provider",
+				Err:    ValidateSupportedProvider(v.provider),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name: "create preflight validations pass",
+				Err:  v.cmdValidations.PreflightValidations(ctx),
```

Comment:
this is running all the validations in `PreflightValidations` as one validation here
Could we just register all those validations individually here?
We might not be able to do this for everything (specially providers) without a big refactor, but this one should be using the same types for validations, right?

Reply:
I think this would be good to do, but it would also make `validate create cluster` more independent of `create cluster` since `create cluster` is still reliant on the `createValidations.PreflightValidations()`. 

Could this be done in a separate PR, maybe accompanying  `create cluster` using this or an updated set of validations?

Reply:
Not necessarily. We can still have all the validations defined where they are. Just allow to return a slice with all validations as opposed to only have one method that returns them